File: src/main/java/com/rs/java/game/area/banks/VarrockWestBank.java
package com.rs.java.game.area.banks;

import com.rs.java.game.WorldTile;
import com.rs.java.game.area.Area;
import com.rs.java.game.area.Shape;
import com.rs.java.game.area.shapes.Polygon;

public final class VarrockWestBank extends Area {

	@Override
	public Area update() {
		return this;
	}

	@Override
	public String name() {
		return "Varrock West Bank";
	}

	public VarrockWestBank() {
		super(
				new Polygon(new WorldTile[] {
				new WorldTile(3179, 3432, 0), new WorldTile(3179, 3446, 0),
				new WorldTile(3194, 3446, 0), new WorldTile(3194, 3432, 0) }) );
	}

	@Override
	public boolean member() {
		return false;
	}

	@Override
	public Environment environment() {
		return Environment.SAFEZONE;
	}

}

File: src/main/java/com/rs/kotlin/game/world/area/areas/banks/VarrockWestBank.kt
package com.rs.kotlin.game.world.area.areas.banks

import com.rs.java.game.WorldTile
import com.rs.kotlin.game.world.area.Area
import com.rs.kotlin.game.world.area.shape.Polygon

class VarrockWestBank : Area(
    Polygon(
        arrayOf(
            WorldTile(3179, 3432, 0), WorldTile(3179, 3446, 0),
            WorldTile(3194, 3446, 0), WorldTile(3194, 3432, 0)
        )
    )
) {
    override fun update(): Area {
        return this
    }

    override fun name(): String {
        return "Varrock West Bank"
    }

    override fun member(): Boolean {
        return false
    }

    override fun environment(): Environment {
        return Environment.SAFEZONE
    }
}
